src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,338 @@
+                if (performSubstitutions) {
+                    transformedValue = performSubstitutions(transformedValue);
+                }
+
+                // Re-assemble the CSV line and place in buffer.
+                String[] csvValues = csvReader.getValues();
+                csvValues[transformFieldIndex] = transformedValue;
+
+                // Write line to table (plus new line char).
+                writer.write(csvValues);
+
+                // Count number of CSV rows changed.
+                if (!originalValue.equals(transformedValue)) {
+                    modifiedRowCount++;
+                }
+            } // End of iteration over each row.
+            csvReader.close();
+            writer.flush();
+            Long stopMillis = System.currentTimeMillis();
+
+            // Copy csv input stream into the zip file, replacing the existing file.
+            try (
+                // Modify target zip file that we just read.
+                FileSystem targetZipFs = FileSystems.newFileSystem(tempZipPath, null);
+                // Stream for file copy operation.
+                InputStream inputStream =  new ByteArrayInputStream(stringWriter.toString().getBytes(StandardCharsets.UTF_8))
+            ) {
+                Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
+                Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
+                zipTarget.feedTransformResult.tableTransformResults.add(
+                    new TableTransformResult(tableName, 0, modifiedRowCount, 0)
+                );
+            }
+
+            // Replace original zip file with temporary working zip file.
+            // (This should also trigger a system IO update event, so subsequent IO calls pick up the correct file.
+            Files.move(tempZipPath, originalZipPath, StandardCopyOption.REPLACE_EXISTING);
+            Long transformEndMillis = System.currentTimeMillis();
+            LOG.info(
+                "Field normalization transformation successful: {} row(s) changed in {} ms ({} ms including file ops).",
+                modifiedRowCount,
+                stopMillis - startMillis,
+                transformEndMillis - transformStartMillis
+            );
+        } catch (Exception e) {
+            status.fail("Unknown error encountered while transforming zip file", e);
+        }
+    }
+
+    /**
+     * Converts the provided string to Title Case,
+     * accommodating for separator characters that may be immediately precede
+     * (without spaces) names that we need to capitalize.
+     */
+    public String convertToTitleCase(String inputString) {
+        // Return at least a blank string.
+        if (StringUtils.isBlank(inputString)) {
+            return "";
+        }
+
+        String result = WordUtils.capitalizeFully(inputString, SEPARATORS);
+
+        // Exceptions (e.g. acronyms) should remain capitalized.
```

Comment:
These can also be words that should remain lowercase (or their own special mix of cases), right? Maybe the comment should be updated for that. Same goes for method javadoc.

Reply:
Yes, there is a test case in `NormalizeFieldTransformationTest`, I'll also add the comments in there.